flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/DataStatisticsAndRecordWrapper.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink.shuffle;
+
+import java.io.Serializable;
+import org.apache.iceberg.flink.sink.shuffle.statistics.DataStatistics;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * The wrapper class for data statistics and record. It is the only way for data statistics operator to send
+ * global data statistics to custom partitioner to distribute data based on statistics
+ *
+ * <p>DataStatisticsAndRecordWrapper is sent from {@link DataStatisticsOperator} to partitioner. It
+ * contains either data statistics(globally aggregated) or a record. Once partitioner receives the data
+ * statistics, it will use that to decide the coming record should send to which writer subtask. After
+ * shuffling, a filter and mapper are required to filter out the data distribution weight, unwrap the
+ * object and extract the original record type T.
+ */
+public class DataStatisticsAndRecordWrapper<T, K> implements Serializable {
```

Comment:
@yegangy0718  `ShuffleRecord` probably still makes sense for this wrapper class.

I saw some other classes in Iceberg with names like `FooAndBar`. But they typically contains both `Foo` **and** `Bar`. Here is  an `or` relationship. Hence I don't know if `And` is the most accurate name. 

Regardless, I prefer us remove the `Wrapper` suffice from the name. it can be described in the Javadoc

Reply:
The reason I rename it is, in `DataStatisticsOperator`, it generates the object which contains either global data statistics or record, while the place where shuffle happens is at the partitioner. The name `DataStatisticsAndRecord` is closer to the usage of the class(transmit global data statistics).

Indeed, like you said, the other class which uses `And` contains both. For example, `RecordAndPosition` class, it contains both record and position. 
How about naming it `DataStatisticsOrRecord`  even though there is no such kind of `or` class in the current repo :( 
WDYT

Reply:
`DataStatisticsOrRecord` sounds good to me